apps/web/vibes/soul/form/file-input/file-dropzone.tsx

```diff
@@ -0,0 +1,323 @@
+
+      const dataTransfer = new DataTransfer();
+      for (const file of pastedFiles) {
+        dataTransfer.items.add(file);
+      }
+
+      inputElement.files = dataTransfer.files;
+      inputElement.dispatchEvent(new Event('change', { bubbles: true }));
+    },
+    [inputRef, onPasteProp],
+  );
+
+  const onKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      onKeyDownProp?.(event);
+
+      if (!event.defaultPrevented && (event.key === 'Enter' || event.key === ' ')) {
+        event.preventDefault();
+        inputRef.current?.click();
+      }
+    },
+    [inputRef, onKeyDownProp],
+  );
+
+  const onClick = useCallback(
+    (event: MouseEvent<HTMLButtonElement | HTMLDivElement>) => {
+      if (event.defaultPrevented) return;
+
+      inputRef.current?.click();
+    },
+    [inputRef],
+  );
+
+  const onChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const files = Array.from(event.target.files ?? []);
+      onFilesAdded(files);
+
+      if (inputRef.current) {
+        inputRef.current.value = '';
+      }
+    },
+    [onFilesAdded],
+  );
+
+  return (
+    <>
+      <Label className={clsx(hideLabel === true ? 'sr-only' : 'mb-2')} htmlFor={labelId}>
+        {label}
+      </Label>
+      <div
+        aria-controls={`${inputId} ${listId}`}
+        aria-disabled={disabled}
+        className={clsx(
+          'flex min-h-19 cursor-pointer flex-col items-center justify-center gap-1.5 rounded-lg border-[1.5px] border-dashed border-(--file-input-dropzone-border,var(--contrast-200)) bg-(--file-input-dropzone-background,var(--background)) p-4 transition-colors duration-300 ease-in-out select-none @sm:flex-row',
+          // Hover styles
+          'hover:border-(--file-input-dropzone-border-hover,var(--foreground)) hover:bg-(--file-input-dropzone-background-hover,var(--contrast-100))',
+          // Focus styles
+          'focus-visible:ring-2 focus-visible:ring-(--file-input-focus,var(--primary)) focus-visible:ring-offset-2 focus-visible:outline-hidden',
+          // Disabled styles
+          'data-[disabled]:cursor-not-allowed data-[disabled]:border-(--file-input-dropzone-border-disabled,color-mix(in_oklab,var(--contrast-200)_70%,transparent)) data-[disabled]:bg-(--file-input-dropzone-background-disabled,var(--background)) data-[disabled]:focus-visible:ring-0 data-[disabled]:focus-visible:ring-offset-0 data-[disabled]:focus-visible:outline-none',
+          // Invalid styles
+          'data-[invalid]:border-(--file-input-dropzone-border-error,var(--error))',
+          // Dragging styles
+          'data-[dragging]:border-(--file-input-dropzone-border-dragging,var(--foreground)) data-[dragging]:bg-(--file-input-dropzone-background-dragging,var(--success-highlight))',
+        )}
+        data-disabled={disabled === true ? '' : undefined}
+        data-dragging={isDragging ? '' : undefined}
+        data-invalid={invalid === true ? '' : undefined}
```

Comment:
I like the fact that you broke up tailwind styles by pseudo state. I've started doing that and I think it's helpful!